taskboard/backend_interface.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict, Any
+
+class TaskBoardBackend(ABC):
```

Comment:
I'd say this is the `TaskBoard`'s interface.
It should simply be called `TaskBoard`.